lib/overcommit/hook/pre_commit/hard_tabs.rb

```diff
@@ -9,5 +9,14 @@ def run
         /^(?<file>(?:\w:)?[^:]+):(?<line>\d+)/,
       )
     end
+
+    def applicable_files
+      # Filter out `Makefile`, `makefile`, and `*.go` by default, since hard
+      # tabs are *required* in all these languages
+      super.reject do |modified|
+        fname = File.basename(modified)
+        fname =~ /Makefile/i or fname.end_with? '.go'
```

Comment:
Would it make sense to be a little more strict on the makefile regexp? So that we don't catch things like `makeFileArray.js` etc.